spec/07-variables.md

```diff
@@ -202,7 +202,7 @@ calls.
 
 **Constraints**
 
-Each *expression* must designate a variable name.
+Each *expression* must designate a variable name that is not the result of the non-brace form of [variable-name-creation-expression](10-expressions#variable-name-creation-operator).
```

Comment:
@RexJaeschke Looks like meaning got lost in translation from RFC text to user documentation. "Simple variable" is a compiler term we use to refer to variables of the form `$foo`, `$$foo` and `${$foo}`, i.e. anything designating a real variable, rather than a property or array dimension.

The actual change that happened here is that things like `$$foo->bar` or `$$foo['bar']` are not supported anymore and instead explicit curly braces must be used, i.e. `${$foo->bar}` or `${$foo['bar']}`.